src/plugins/enum/enum.c

```diff
@@ -42,11 +43,9 @@ int elektraEnumGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned ELEKTRA_UN
 	return 1; /* success */
 }
 
-static int validateKey (Key * key)
+static int validateWithList (Key * key)
 {
-	const Key * meta = keyGetMeta (key, "check/enum");
-	if (!meta) return 1;
-	const char * validValues = keyString (meta);
+	const char * validValues = keyString (keyGetMeta (key, "check/enum"));
```

Comment:
Yes, it sounds necessary. It could even contain the separator.

Btw. we could write a convenience-shell tool that can split up arguments as array, e.g.:

`kdb setmeta user/abc check/enum -a "first enum" "second enum" "third enum"`